qutebrowser/app.py

```diff
@@ -343,15 +343,12 @@ def on_focus_changed(_old, new):
         log.misc.debug("on_focus_changed called with non-QWidget {!r}".format(
             new))
 
-    if new is None or not isinstance(new, mainwindow.MainWindow):
-        try:
-            objreg.delete('last-focused-main-window')
-        except KeyError:
-            pass
-        qApp.restoreOverrideCursor()
-    else:
-        objreg.register('last-focused-main-window', new.window(), update=True)
+    if new is not None:
```

Comment:
Hmm, though I see another issue here: `qapp.activeWindow()` could actually be the web inspector, or a dialog box, or another top-level window... And apparently we can't get the toplevel `QWidget` from a `QWindow`?

So I guess the real solution would be to get [`new.window()`](http://doc.qt.io/qt-5/qwidget.html#window) (which is actually the toplevel `QWidget`, fun...) and register that if it's a `mainwindow.MainWindow` instance.